tensorflow/core/platform/default/logging.cc

```diff
@@ -367,6 +364,14 @@ bool LogEveryNSecState::ShouldLog(double seconds) {
   return true;
 }
 
+TFLogSinks::TFLogSinks()
+{
+#ifndef NO_DEFAULT_LOGGER
+  static TFDefaultLogSink default_sink;
+  sinks_.push_back(&default_sink);
```

Comment:
I disagree, `TFLogSinks::Add` does a whole lot more than just pushing back a log sink. Since this is the constructor, we know that it's thread safe and there are no sinks, or messages in the queue.